extensions/database/module/scripts/index/database-source-ui.js

```diff
@@ -91,24 +91,44 @@ Refine.DatabaseSourceUI.prototype.attachUI = function(body) {
   
   
   this._elmts.databaseTypeSelect.change(function(event) {
-      var type = $( "#databaseTypeSelect" ).val(); 
-      if(type === "postgresql"){ 
-          $( "#databaseUser" ).val("postgres");
-          $( "#databasePort" ).val("5432");	  
-      }else if(type === "mysql"){	  
-          $( "#databaseUser" ).val("root");
-          $( "#databasePort" ).val("3306");	  
-      }else if(type === "mariadb"){
-          $( "#databaseUser" ).val("root");
-          $( "#databasePort" ).val("3306");
-      }else if(type === "sqlite"){
-          $( "#databaseUser" ).val("na");
-          $( "#databasePort" ).val("0");
-          $( "#databaseHost" ).val("na");
-      }else{
-          $( "#databaseUser" ).val("root");
-          $( "#databasePort" ).val("3306");
-      }
+    var type = $( "#databaseTypeSelect" ).val();
+    var databaseClass = "";
+  
+    if(type === "postgresql") { 
+        $( "#databaseUser" ).val("postgres");
+        $( "#databasePort" ).val("5432");
+        databaseClass = "psgsq"
+      
+    } else if(type === "mysql") {	  
+        $( "#databaseUser" ).val("root");
+        $( "#databasePort" ).val("3306");	  
+        databaseClass = "mysql"
+      
+    } else if(type === "mariadb") {
+        $( "#databaseUser" ).val("root");
+        $( "#databasePort" ).val("3306");
+        databaseClass = "mardb"
+      
+    } else if(type === "sqlite") {
+        $( "#databaseUser" ).val("na");
+        $( "#databasePort" ).val("0");
+        $( "#databaseHost" ).val("na");
+        databaseClass = "sqlit"
```

Comment:
Is there a particular reason not to reuse the `type` directly as class (perhaps with a prefix, to be on the safe side)? That would simplify the code a bit. You would just have one line `databaseClass = "dbtype" + type`.

Reply:
> Is there a particular reason not to reuse the `type` directly as class (perhaps with a prefix, to be on the safe side)? That would simplify the code a bit. You would just have one line `databaseClass = "dbtype" + type`.

Yes. The full name would be fine, just 2x longer. I also though of just have TWO classes network_db and file_db. It would even more simply the code, but wouldn't permit to have specific fields per database, just between database types.

So, I go for the full name?

Regards,
   Antoine

Reply:
> Is there a particular reason not to reuse the `type` directly as class (perhaps with a prefix, to be on the safe side)? That would simplify the code a bit. You would just have one line `databaseClass = "dbtype" + type`.

@wetneb: It's done. You can review it again.

Regards, A.

Reply:
> Is there a particular reason not to reuse the `type` directly as class (perhaps with a prefix, to be on the safe side)? That would simplify the code a bit. You would just have one line `databaseClass = "dbtype" + type`.

Hi @weblate; I wrote too fast when I said I did what you asked. I forgot half of what you asked! ;-) So here's the other half. For the class name `options`, I changed it to: `dbtype-options`. For the name of the db, I took the full name like you asked, but only prefixed it with `dbt-`. Some names are long, and imagine a user with 10 databases type (core-OR + extensions), it would make a long class attribute value… Anyway, it drastically reduce the risk of collisions. 

Regards, A.